### PR TITLE
Add initial query page

### PR DESCRIPTION
**Pull request by Arif-Khalid**
### Changes made
* Added query page containing placeholder app logo
* Used controlled input behaviour on query input
* Added prettier package and put prettier config in .prettierrc file
* Add format command to package.json
* Add simple grid styling to content
* Add temporary fake API call returning fake data when query sends

###
Before sending query

After sending query

### CODE REVIEW

**Review by nknguyenhc — COMMENTED** [review]
@Arif-Khalid Is it possible for you to add the `.prettierrc` file, and possibly an `npm` command to do style correction?

**Review by nknguyenhc — COMMENTED** [review]
Also, I believe file naming convention should be spinal case, e.g. `content-component`, `query-component`.

**Comment by Arif-Khalid** [comment]
> Also, I believe file naming convention should be spinal case, e.g. `content-component`, `query-component`.

Renamed appropriately

> @Arif-Khalid Is it possible for you to add the `.prettierrc` file, and possibly an `npm` command to do style correction?

Added format command to package.json and moved prettier config to .prettierrc

**Comment by Arif-Khalid** [comment]
After the most recent review I have also added simple grid styling, a button that sends the query and a custom hook that calls a simulated fetch.

File: frontend/src/hooks/useProducts.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      //TODO add error handling
+      return
+    }
+
+    // Fake API call
+    setTimeout(() => {
+      setProducts(testProducts)
+      alert(`Query Received: ${query}`)
+    }, 1000)
+  }
```
Comment (nknguyenhc) [inline frontend/src/hooks/useProducts.ts]: Better to make nested functions `useCallback` hooks. The `useCallback` hooks allow the function definition to cached between renders, while function declaration refreshes the function declaration on every re-render.

File: frontend/src/components/query-component/query-component.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          setQuery(event.target.value)
+        }}
+        placeholder="Describe the product"
+        disableUnderline={true}
+      ></Input>
+      <IconButton
+        onClick={() => {
+          sendQuery(query)
+          setQuery('')
+        }}
```
Comment (nknguyenhc) [inline frontend/src/components/query-component/query-component.tsx]: Should abstract out the callbacks as hooks. Definition of a listener declared with a lambda expression is never refreshed, while definition of a listener with a hook is refreshed when necessary.

```ts
export function QueryComponent({ sendQuery }: QueryComponentProps) {
  const handleClick = useCallback(() => {
    sendQuery();
    setQuery('');
  }, [sendQuery]);

  // Other parts

  return (
    // ...
    <IconButton
      onClick={handleClick}
      className="query-send-icon-button"
      aria-label="send"
    />
    // ...
  );
}
```

**Review by nknguyenhc — COMMENTED** [review]
A few comments on the UI:

image

1. The screen before searching should not have any scroll bar. This can be done by setting the `overflow` property to `overflow: auto` instead of `overflow: scroll`.
2. User should be able to press enter to do the search, instead of having to press the send button. I would suggest wrapping the input field in a `form` element, because the `form` element handles hitting enter for you. But up to you how you want to resolve.

Besides, some comment on quality of code below.

File: frontend/src/hooks/useProducts.ts (hunk 1 above)
Comment (Arif-Khalid) [reply frontend/src/hooks/useProducts.ts]: Good call I have refactored to use useCallback

File: frontend/src/components/query-component/query-component.tsx (hunk 2 above)
Comment (Arif-Khalid) [reply frontend/src/components/query-component/query-component.tsx]: I agree I have abstracted to a memoized function appropriately

**Comment by Arif-Khalid** [comment]
> A few comments on the UI:
> 
> image
> 
> 1. The screen before searching should not have any scroll bar. This can be done by setting the `overflow` property to `overflow: auto` instead of `overflow: scroll`.
> 2. User should be able to press enter to do the search, instead of having to press the send button. I would suggest wrapping the input field in a `form` element, because the `form` element handles hitting enter for you. But up to you how you want to resolve.
> 
> Besides, some comment on quality of code below.

Good observations which have been addressed by my latest commits

**Review by nknguyenhc — APPROVED** [review]
LGTM!

**Pull request merged**